### PR TITLE
chore: 🤖 silence `deprecate_early_static` warnings

### PR DESCRIPTION
**Pull request by DhariniJeeva**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10577)

this pr will silence the `deprecate_early_static` [warning](https://rfcs.emberjs.com/id/0741-ember-data-deprecate-model-static-field-access-without-lookup/), which affects only when mocks are used. This is due to the discoverEmberDataModels method that accesses the static fields of the model directly. 

Screenshot 2023-08-15 at 11 02 49 AM

### CODE REVIEW

File: addons/core/addon/initializers/deprecations.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ import { registerDeprecationHandler } from '@ember/debug';
 
 export function initialize() {
   registerDeprecationHandler((message, options, next) => {
+    //silence deprecate-early-static warnings till we find a way to
```
Comment (ZedLi) [inline addons/core/addon/initializers/deprecations.js:10]: Maybe it would also help to add a TODO as part of the comment?

**Review by ZedLi — APPROVED** [review]
Thanks for taking care of this!

File: addons/core/addon/initializers/deprecations.js (hunk 1 above)
Comment (DhariniJeeva) [reply addons/core/addon/initializers/deprecations.js:11]: done

**Review by lisbet-alvarez — APPROVED** [review]

**Review by calcaide — APPROVED** [review]
Nice!! thanks for looking into this!!

**Pull request merged**